micro_sam/util.py

```diff
@@ -200,16 +178,23 @@ def _available_devices():
 
 
 def get_sam_model(
-    model_type: str = _DEFAULT_MODEL,
+    model_name: str = _DEFAULT_MODEL,
     device: Optional[Union[str, torch.device]] = None,
     checkpoint_path: Optional[Union[str, os.PathLike]] = None,
     return_sam: bool = False,
 ) -> SamPredictor:
     r"""Get the SegmentAnything Predictor.
 
-    This function will download the required model checkpoint or load it from file if it
-    was already downloaded.
-    This location can be changed by setting the environment variable: MICROSAM_CACHEDIR.
+    This function will download the required model or load it from the cached weight file.
+    This location of the cache can be changed by setting the environment variable: MICROSAM_CACHEDIR.
+    The name of the requested model can be set via `model_name`.
+    See https://computational-cell-analytics.github.io/micro-sam/micro_sam.html#finetuned-models
+    for an overview of the available models
+
+    Alternatively this function can also load a model from weights stored in a local filepath.
+    The corresponding file path is given via `checkpoint_path`. In this case `model_name`
+    must be given as the matching encoder architecture, e.g. "vit_b" if the weights are for
+    a SAM model with vit_b encoder.
```

Comment:
I added some explanation to try to make clear what happens depending on passing a `checkpoint_path` or not. Let me know if this is still unclear and what could be improved.

Reply:
This helps - I think the key point is that checkpoint_path is only for **local** weights. I was imagining a situation where people would share their finetuned weights and users could download them from zenodo or similar (but I guess that's what the `get_custom_sam_model` function is for)

Reply:
> This helps - I think the key point is that checkpoint_path is only for local weights.

:+1:

> I was imagining a situation where people would share their finetuned weights and users could download them from zenodo or similar (but I guess that's what the get_custom_sam_model function is for)

For now the checkpoint would need to be downloaded for this. In the future we want to support uploading models to bioimage.io and this is how we would support uploading and downloading other finetuned models.
(But we are still waiting for some changes on the bioimage.io for this.)